src/language/extension-types.md

```diff
@@ -0,0 +1,315 @@
+Extension type member declarations are identical to class members.
+Members can be methods, getters, setters, or operators:
+
+```dart
+extension type NumberE(int value) {
+  // Operator:
+  NumberE operator +(NumberE other) =>
+      NumberE(value + other.value);
+  // Getter:
+  NumberE get i => value;
+  // Method:
+  bool isValid() => !value.isNegative;
+}
+```
+
+Members of the representation type are not available to the extension type
+by default. If you want a member from the representation type to be available
+to the extension type, you must write a declaration for it
+in the extension type definition, like the `operator +` in `NumberE`.
+You also can define new members unrelated to the representation type,
+like the `i` getter and `isValid` method.
+
+Extension types can't declare instance fields (unless [`external`][]),
+or [abstract members][].
+
+### Implements
+
+You can optionally use the `implements` clause to introduce
+a subtype relationship on an extension type, making *all* members
+of the representation type available to the extension type:
+
+```dart
+extension type NumberI(int i) 
+  implements int{
+  // ...
+}
+```
+
+An extension type can only implement its representation type
+or a supertype of its representation type.
+
+This is technically NOT an inheritance relationship
+like the one `implements` introduces for classes,
+but rather an [applicability][] relationship like that between an
+extension method and its `on` type.
+Members that apply to the supertype are applicable to the subtype,
+unless the supertype has a more specific definition for the member.
+
+#### `@redeclare`
+
+Replacing a method that shares a name with a member of the supertype
+is not an override relationship for extension types, but a *redeclaration*.
+Use the `@redeclare` annotation to tell the compiler you are *knowingly* choosing
+to use the same name as a supertype's member definition, not just accidentally
+using the same name:
+
+```dart
+class C {
+  void m() {}
+}
+
+extension type E(C c) implements C {
+  @redeclare
+  void m() {}
+}
+```
+
+If you don't use the annotation, the compiler will choose the [more specific][]
+definition wherever the name is invoked, which might not be the behavior you expect.
+
+## Usage
+
+To use an extension type, create an instance the same as you would with a class:
+
+```dart
+extension type NumberE(int value) {
+  NumberE operator +(NumberE other) =>
+      NumberE(value + other.value);
+
+  NumberE get i => value;
+  bool isValid() => !value.isNegative;
+}
+
+void testE() { 
+  var num = NumberE(1);
+}
+```
+
+Then you can invoke members on the object as you would with a class object.
+Any attempt to treat the extension type object as an object
+of the representation type, or vice versa, will result in compile time errors
+(unless that behavior is explicitly defined for the extension type,
+or it is [transparent](#transparency)).
```

Comment:
Not sure how it can be explicitly defined *other* than being "transparent", so this paragraph confuses me.

Consider (if not taking the suggestion above):
> (unless the extension type implements the representation type, then it can be used as having the representation type).